p4-16/spec/P4-16-spec.mdk

```diff
@@ -3313,6 +3315,20 @@ type. The result always has the same width as the left operand.
   - all result bits are zero when shifting left
   - all result bits are zero when shifting right a positive value
   - all result bits are one when shifting right a negative value
+- Extraction of a set of contiguous bits, also known as a slice, denoted by `[m:l]`,
+  where `m` and `l` must be positive integers
+  that are compile-time known values, and `m >= l`. The result is
+  an unsigned bit-string of width `m - l + 1`, including the bits numbered
+  from `l` (which becomes the least significant bit of the result) to `m` (the
+  most significant bit of the result) from the source operand. The conditions `0 <= l < W`
+  and `l <= m < W` are checked statically (where `W` is
+  the length of the source bit-string). Note that both endpoints of
+  the extraction are inclusive. The bounds are required to be
+  compile-time known values so that the result width can be computed
+  at compile time. Slices are also l-values, which means that P4 supports assigning to a slice: ` e[m:l] = x `.
```

Comment:
If the intent is that in the assignment `e[m:l] = expression;` that `expression` must be an unsigned integer type is that worth saying explicitly?  That is, just like the assignment `x = y;` is a compile-time error if x is unsigned and y is signed, so should `e[m:l] = y;` if `e` and `y` are both signed, because `e[m:l]` is an unsigned l-value?

If the answer is yes, then not that this is even a compile-time error, but I don't see a strong reason to include that in the spec.  It probably should be a p4c test case:
```
int<8> signed_int1;
int<8> signed_int2;

signed_int1[7:0] = signed_int2;
```

Reply:
```
./p4test signed.p4 
signed.p4(5): [--Werror=type-error] error: AssignmentStatement: Cannot unify int<8> to bit<8>
        signed_int1[7:0] = signed_int2;
                         ^
```